src/systemctl/systemctl-show.c

```diff
@@ -1989,6 +1989,27 @@ static int print_property(const char *name, const char *expected_value, sd_bus_m
                         if (r < 0)
                                 return bus_log_parse_error(r);
 
+                        return 1;
+                } else if (streq(name, "ExtraFileDescriptorNames")) {
+                        _cleanup_free_ char *extra_fd_names = NULL;
+                        const char *fdname;
+
+                        r = sd_bus_message_enter_container(m, SD_BUS_TYPE_ARRAY, "s");
+                        if (r < 0)
+                                return bus_log_parse_error(r);
+
+                        while ((r = sd_bus_message_read_basic(m, 's', &fdname)) > 0)
+                                if (!strextend_with_separator(&extra_fd_names, " ", fdname))
+                                        return log_oom();
+                        if (r < 0)
+                                return bus_log_parse_error(r);
+
+                        r = sd_bus_message_exit_container(m);
+                        if (r < 0)
+                                return bus_log_parse_error(r);
```

Comment:
just use sd_bus_message_read_strv(), and replace the whole thing here. Then do strv_join() to output the resulting array